compiler/noirc_frontend/src/elaborator/scope.rs

```diff
@@ -28,9 +28,9 @@ impl Elaborator<'_> {
         ModuleId { krate: self.crate_id, local_id: self.local_module() }
     }
 
-    pub fn replace_module(&mut self, new_module: ModuleId) -> ModuleId {
-        let current_module = self.module_id();
-
+    pub fn replace_module(&mut self, new_module: ModuleId) -> Option<ModuleId> {
+        let current_module =
```

Comment:
Was this a bug fix?

Reply:
No. The initial module is empty (well, `local_module`) and I wanted to use this function to set the initial module. However, that method calls `self.module_id()` which panics if there's no module set.

I initially created a separate method for this called `set_module` that just set the module and didn't return anything, not checking the current module. But... I thought it was confusing having two methods called `set_module` and `replace_module` so I just changed `replace_module` to not expect a current module. After all the caller just uses the return value to set it back afterwards.